test/e2e/scenario/recorder.scenario.ts

```diff
@@ -56,4 +63,84 @@ describe('recorder', () => {
         )
       ).toBeTruthy('have a IncrementalSnapshot/MouseInteraction record')
     })
+
+  fdescribe('snapshot', () => {
+    createTest('obfuscate blocks')
+      .withSetup(bundleSetup)
+      .withRumRecorder()
+      .withBody(
+        html`
+          <p id="foo">foo</p>
+          <p id="bar" data-dd-privacy="hidden">bar</p>
+          <p id="baz" class="dd-privacy-hidden">baz</p>
+        `
+      )
+      .run(async ({ events }) => {
+        await flushEvents()
+
+        expect(events.sessionReplay.length).toBe(1)
+        const { segment } = events.sessionReplay[0]
+
+        const fullSnapshots = segment.data.records.filter((record) => record.type === RecordType.FullSnapshot)
+        expect(fullSnapshots.length).toBe(1)
+
+        const documentNode = (fullSnapshots[0] as FullSnapshotRecord).data.node as DocumentNode
+        expect(documentNode.childNodes.length).toBe(2) // DocumentType + html Element
+
+        const htmlNode = documentNode.childNodes[1] as ElementNode
+        expect(htmlNode.childNodes.length).toBe(3) // head Element + Text + body Element
+
+        const bodyNode = htmlNode.childNodes[2] as ElementNode
+        expect(bodyNode.childNodes.length).toBe(7) // 3 p Elements + 4 Texts
+
+        const pFooNode = bodyNode.childNodes[1] as ElementNode
+        expect(pFooNode.childNodes.length).toBe(1)
+        expect((pFooNode.childNodes[0] as TextNode).textContent).toBe('foo')
+
+        const pBarNode = bodyNode.childNodes[3] as ElementNode
+        expect(pBarNode.childNodes.length).toBe(0) // obfuscated
+
+        const pBazNode = bodyNode.childNodes[5] as ElementNode
+        expect(pBazNode.childNodes.length).toBe(0) // obfuscated
```

Comment:
That is a lot of checks and details about the data structure.
could it be interesting/possible to have something like:
```ts
const { segment } = events.sessionReplay[0]
expect(findNodeWithId(segment, 'foo').textContent).toBe('foo')
expect(findNodeWithId(segment, 'bar').textContent).toBe('')
expect(findNodeWithId(segment, 'baz').textContent).toBe('')
```